source/extensions/load_balancing_policies/common/load_balancer_impl.cc

```diff
@@ -637,18 +640,43 @@ absl::FixedArray<ZoneAwareLoadBalancerBase::LocalityPercentages>
 ZoneAwareLoadBalancerBase::calculateLocalityPercentages(
     const HostsPerLocality& local_hosts_per_locality,
     const HostsPerLocality& upstream_hosts_per_locality) {
-  uint64_t total_local_hosts = 0;
-  std::map<envoy::config::core::v3::Locality, uint64_t, LocalityLess> local_counts;
+  std::map<envoy::config::core::v3::Locality, uint64_t, LocalityLess> local_weights;
+  std::map<envoy::config::core::v3::Locality, uint64_t, LocalityLess> upstream_weights;
+  uint64_t total_local_weight = 0;
   for (const auto& locality_hosts : local_hosts_per_locality.get()) {
-    total_local_hosts += locality_hosts.size();
+    uint64_t locality_weight = 0;
+    // If locality_basis_ is set to HOST_WEIGHT, we use the host's weight to calculate the locality
+    // percentage. Otherwise, we count the number of hosts in the locality.
+    if (locality_basis_ == LocalityLbConfig::ZoneAwareLbConfig::HOST_WEIGHT) {
```

Comment:
In case someone adds another field in the enum, we want to make sure we cover all cases. If it were still a boolean, this would be appropriate, but now we have a potential hiding spot for a bug in the future. Can you just do something like:


```suggestion
    switch (locality_basis_) {
      case LocalityLbConfig::ZoneAwareLbConfig::HOST_WEIGHT:
        for (const auto& host : locality_hosts) {
          locality_weight += host->weight();
        }
        break;
      case LocalityLbConfig::ZoneAwareLbConfig::NUM_HEALTHY_HOSTS:
        locality_weight += host->weight();
        break;
      default:
        PANIC_DUE_TO_CORRUPT_ENUM;
    }
```